native/src/routes/SearchModal.tsx

```diff
@@ -63,6 +65,15 @@ const SearchModal = ({
     closeModal(query)
   }
 
+  if (!searchResults) {
+    return (
+      <Wrapper {...testID('Search-Page')}>
+        <SearchHeader query={query} closeSearchBar={onClose} onSearchChanged={setQuery} />
+        {!!query.length && <Loader />}
+      </Wrapper>
+    )
+  }
```

Comment:
Hmm... I think I am not this happy with this, I think all loading should be handled in the SearchModalContainer. This introduces some kind of duplicated code (and uses a different loading spinner). Could you try to solve this in some other way? Perhaps it is necessary to move the query state one level up to SearchModalContainer for it.

Reply:
It's tricky to move the loading of the results into the SearchModalContainer. For one, moving the query up there leads to a re-render each time the query changes. I guess we could give a loading and isLoading parameter to the SearchModal that then changes the behaviour of the LoadingErrorHandler.

But there is the additional issue that the LoadingErrorHandler handles loading the entire page. We specifically want to already show part of the page (the search bar), even while the results are still being loaded into minisearch. If we were to handle all the loading in the LoadingErrorHandler, then the user would be stuck with the white screen again for longer, even though we can perfectly well show the SearchBar already.

Reply:
Thanks for the explanation, makes sense! Then my only request would be to style the activity indicator to match the style of the refresh control (which probably internally uses the activity indicator component) that we use all over the app, e.g. in the LoadingErrorHandler, if possible :)

Reply:
But actually, this does not have to be done here. I just think it would be nice to have it look uniform. I'll create a new issue.

Reply:
Good point, I'll add that here :)